src/test/com/Elessar/app/client/MyClientIntegTest.java

```diff
@@ -0,0 +1,233 @@
+                assertEquals(true, cursor.hasNext());
+            }
+
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.USERS).find(eq(User.NAME, "Zhi Xu")).iterator()) {
+                assertEquals(true, cursor.hasNext());
+            }
+
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+            assert(false);
+        }
+    }
+
+
+    private void logOnTest() {
+        try {
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.USERS).find(eq(User.NAME, "Shuai Hu")).iterator()) {
+                if (cursor.hasNext()) {
+                    assertEquals(false, cursor.next().getBoolean(User.ONLINE));
+                }
+            }
+
+            LogonResponse logonResp1 = clientA.logOn("Ertai Cai", "2222123", clientA_Port);
+            Assert.assertEquals(false, logonResp1.getSuccess());
+
+            LogonResponse logonResp2 = clientA.logOn("Shuai Hu", "4238903", clientA_Port);
+            Assert.assertEquals(false, logonResp2.getSuccess());
+
+            LogonResponse logonResp3 = clientA.logOn("Shuai Hu", "4238902", clientA_Port);
+            Assert.assertEquals(true, logonResp3.getSuccess());
+
+            LogonResponse logonResp4 = clientA.logOn("Shuai Hu", "4238902", clientA_Port);
+            Assert.assertEquals(true, logonResp4.getSuccess());
+
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.USERS).find(eq(User.NAME, "Shuai Hu")).iterator()) {
+                if (cursor.hasNext()) {
+                    assertEquals(true, cursor.next().getBoolean(User.ONLINE));
+                }
+            }
+
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+            assert(false);
+        }
+    }
+
+
+    private void logOffTest() {
+        try {
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.USERS).find(eq(User.NAME, "Shuai Hu")).iterator()) {
+                if (cursor.hasNext()) {
+                    assertEquals(true, cursor.next().getBoolean(User.ONLINE));
+                }
+            }
+
+            LogoffResponse logoffResp1 = clientA.logOff("Ertai Cai");
+            Assert.assertEquals(false, logoffResp1.getSuccess());
+
+            LogoffResponse logoffResp2 = clientA.logOff("Zhi Xu");
+            Assert.assertEquals(true, logoffResp2.getSuccess());
+
+            LogoffResponse logoffResp3 = clientA.logOff("Shuai Hu");
+            Assert.assertEquals(true, logoffResp3.getSuccess());
+
+            LogoffResponse logoffResp4 = clientA.logOff("Shuai Hu");
+            Assert.assertEquals(true, logoffResp4.getSuccess());
+
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.USERS).find(eq(User.NAME, "Shuai Hu")).iterator()) {
+                if (cursor.hasNext()) {
```

Comment:
same comment for the "if" here: we should get rid of this if.
also should do the same check for user "Zhi Xu"